git_apple_llvm/am/zippered_merge.py

```diff
@@ -0,0 +1,85 @@
+"""
+  Module for performing a zippered merge between two branches.
+
+  A zippered merge is used to merge branches based on a matching
+  merge-base commit from the common ancestor.
+
+  No date ordering is done, the left branch is merged before the
+  right branch while the merge bases are matching.
```

Comment:
I don't think the scenario is clear enough.  The merge base with what?   Can you add an example?